### DISCUSSION TITLE
Capturing payments from physical cards via Point of Sale (POS) devices (Stripe Terminal, etc)

### DISCUSSION THREAD

**Discussion by rsmithlal** [solidusio/solidus#5238] [Troubleshooting]
Hi folks,

I'm planning to build a digital marketplace for my region. I have been surveying businesses in my area and most would require some way to capture payment from customers in person when they're selling on site at events or at their brick and mortar stores. Is this something that is possible with Solidus in some of the existing plugins such as the Stripe or Paypal payment gateways?

In my research I have found that [Stripe](https://stripe.com/en-ca/terminal) and [Paypal](https://www.paypal.com/us/business/pos-system/card-reader) each have their own POS devices or APIs.

Could a custom payment provider be hypothetically built to interface with these devices and the requisite APIs to complete the payment and order?

Are there any existing projects or plugins using such a thing? My research has not turned up anything so far. I have used the solidus_stripe plugin in the past to take CC payments for online orders.

Thank you kindly!

**Comment by jarednorman** [comment]
People have integrated POS systems with Solidus, though everything I've seen has been pretty specialized to deal with very specific setups. It's not a common use case. Often POS systems integrate directly with ERPs or are totally independent.

**Reply by rsmithlal** [reply]
Thank you for your feedback! Are there any implementations that you know of that are open source or could be viewed as examples of such an integration in a Solidus context?

**Comment by kennyadsl** [comment]
I second what @jarednorman said. In your specific case, why do you want to connect the marketplace's store with the in person payment? Is it for decreasing stocks when items are sold? To keep track of all the orders in a single place? Based on this, you can definitely build an integration with Stripe Terminal or PayPal Card reader, but there's nothing already made that I know about.

**Reply by rsmithlal** [reply]
My particular use case would be using the solidus store as a single or multi-vendor marketplace for artisans to sell their craft goods online. These would be small vendors that don't have another inventory system or physical storefront and would use their e-commerce store to keep track of inventory accurately.

A lot of the vendors attend things like physical marketplaces or craft fairs, and being able to integrate with a POS device or payment terminal would allow them to process in-person payments from their customers with their physical cards instead of having to enter their credit card number. This would reduce sale friction and increase customer confidence in the onsite processing of their payments.

Thank you, I appreciate your feedback.

**Reply by peterberkenbosch** [reply]
Love this use case, actually had a startup build around something similar in the UK. Covid hit and our use case was not practical at the time. I do second what Jared and Alberto are saying. It's not something you can grab off the shelve. Personally I would take a look at Shopify POS and see if that is enough to work with. If I would restart again I would explore stripe pos with the webhook events. Will take some development efforts for sure.
